From a882880205cbaf514160b53df973c7456c763d32 Mon Sep 17 00:00:00 2001
From: Herrie <Github.com@herrie.org>
Date: Fri, 29 Dec 2023 09:55:48 +0100
Subject: [PATCH] Add back various API's

Partial revert of https://github.com/webosose/com.webos.service.location/commit/f9c57af95a15caabbae3fab11bce18e319052d1e

The implementation still seems to be there, however the API calls have been removed/disabled.

Let's simply add them back for testing.

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
---
Upstream-Status: Inappropriate [LuneOS Specific]

 files/sysbus/com.webos.service.location.api.json |  7 +++++++
 src/lunaIpc/LocationService.cpp                  | 12 ++++++++----
 2 files changed, 15 insertions(+), 4 deletions(-)

diff --git a/files/sysbus/com.webos.service.location.api.json b/files/sysbus/com.webos.service.location.api.json
index f9b4e7e..dae475e 100644
--- a/files/sysbus/com.webos.service.location.api.json
+++ b/files/sysbus/com.webos.service.location.api.json
@@ -6,7 +6,13 @@
     "com.webos.service.location/mock/setLocation"
   ],
   "location.query": [
+    "com.webos.service.location/sendExtraCommand",
+    "com.webos.service.location/setGPSParameters",
+    "com.webos.service.location/stopGPS",
+    "com.webos.service.location/geofence/addGeofenceArea",
+    "com.webos.service.location/geofence/removeGeofenceArea",
     "com.webos.service.location/getAllLocationHandlers",
+    "com.webos.service.location/getCachedPosition",
     "com.webos.service.location/getGeoCodeLocation",
     "com.webos.service.location/getGpsSatelliteData",
     "com.webos.service.location/getGpsStatus",
@@ -14,6 +20,7 @@
     "com.webos.service.location/getLocationUpdates",
     "com.webos.service.location/getNmeaData",
     "com.webos.service.location/getReverseLocation",
+    "com.webos.service.location/getTimeToFirstFix",
     "com.webos.service.location/getState"
   ]
 }
diff --git a/src/lunaIpc/LocationService.cpp b/src/lunaIpc/LocationService.cpp
index 36c7142..2b28530 100644
--- a/src/lunaIpc/LocationService.cpp
+++ b/src/lunaIpc/LocationService.cpp
@@ -45,9 +45,13 @@ LSMethod LocationService::rootMethod[] = {
         {"getState",                  LocationService::_getState},
         {"getLocationHandlerDetails", LocationService::_getLocationHandlerDetails},
         {"getGpsSatelliteData",       LocationService::_getGpsSatelliteData},
-//        {"getTimeToFirstFix",         LocationService::_getTimeToFirstFix},
+        {"getTimeToFirstFix",         LocationService::_getTimeToFirstFix},
         {"getLocationUpdates",        LocationService::_getLocationUpdates},
-//        {"getCachedPosition",         LocationService::_getCachedPosition},
+        {"getCachedPosition",         LocationService::_getCachedPosition},
+        {"sendExtraCommand",          LocationService::_sendExtraCommand},
+        {"stopGPS",                   LocationService::_stopGPS},
+        {"exitLocation",              LocationService::_exitLocation},
+        {"setGPSParameters",          LocationService::_setGPSParameters},
         {0,                           0}
 };
 
@@ -242,14 +246,14 @@ bool LocationService::locationServiceRegister(const char *srvcname, GMainLoop *m
 
     bRetVal = LSCategorySetData(*msvcHandle, "/", this, &mLSError);
     LSERROR_CHECK_AND_PRINT(bRetVal, mLSError);
-/*
+
     // add geofence category
     bRetVal = LSRegisterCategoryAppend(*msvcHandle, "/geofence", geofenceMethod, NULL, &mLSError);
     LSERROR_CHECK_AND_PRINT(bRetVal, mLSError);
 
     bRetVal = LSCategorySetData(*msvcHandle, "/geofence", this, &mLSError);
     LSERROR_CHECK_AND_PRINT(bRetVal, mLSError);
-*/
+
     // add mock categoty
     bRetVal = LSRegisterCategoryAppend(*msvcHandle, "/mock", mockPublicMethod, NULL, &mLSError);
     LSERROR_CHECK_AND_PRINT(bRetVal, mLSError);
